vpr/src/noc/bfs_routing.h

```diff
@@ -0,0 +1,80 @@
+#ifndef BFS_ROUTING
+#define BFS_ROUTING
+
+/**
+ * @file 
+ * @brief This file defines the BFSRouting class, which represents a routing algorithm
+ * based on Breadth First Search (BFS).
+ * 
+ * Overview
+ * ========
+ * The BFSRouting class performs routing between routers in the
+ * NoC. This class is based on the BFS algorithm.
+ * 
+ * The BFS algorithm is used to explore the NoC from the starting router in the
+ * traffic flow. Once the destination router is found a path from the source to
+ * the destination router is generated. The main advantage of this algorithm is
+ * that the found path from the source to the destination router uses the 
+ * minimum number of links required within the NoC.
+ * 
+ */
+
+#include <vector>
+#include <unordered_map>
+#include <queue>
+
+#include "noc_routing.h"
+
+class BFSRouting : public NocRouting {
+  public:
+    ~BFSRouting() override;
+
+    /**
+     * @brief Finds a route that goes from the starting router in a 
+     * traffic flow to the destination router. Uses the breadth first search
+     * algorithm to determine the route. A route consists of a series
+     * of links that should be traversed when travelling between two routers
+     * within the NoC.
+     * 
+     * @param src_router_id The source router of a traffic flow. Identifies 
+     * the starting point of the route within the NoC. This is represents
+     * a unique identifier of the source router.
```

Comment:
Same comment as in the base class; mention physical routers and fix typos (if it has the same typos).

Reply:
fixed